openthread/Kconfig

```diff
@@ -81,16 +75,12 @@ config OPENTHREAD_NORDIC_LIBRARY_FTD
 	bool "Nordic Optimized FTD configuration"
 	select OPENTHREAD_COAP
 	select OPENTHREAD_DNS_CLIENT
-	select OPENTHREAD_DHCP6_CLIENT
 	select OPENTHREAD_ECDSA
 	select OPENTHREAD_EXTERNAL_HEAP
 	select OPENTHREAD_IP6_FRAGM
-	select OPENTHREAD_JOINER
 	select OPENTHREAD_NETDIAG_CLIENT
 	select OPENTHREAD_SLAAC
-	select OPENTHREAD_SNTP_CLIENT
 	select OPENTHREAD_SRP_CLIENT
-	select OPENTHREAD_DUA if !OPENTHREAD_THREAD_VERSION_1_1
 	select OPENTHREAD_MLR if !OPENTHREAD_THREAD_VERSION_1_1
 	select OPENTHREAD_LINK_METRICS_INITIATOR if !OPENTHREAD_THREAD_VERSION_1_1
```

Comment:
I believe we can remove `OPENTHREAD_LINK_METRICS_INITIATOR` from FTD as well, the requirement is to support the Subject role only.

Reply:
@LuDuda , if I remember correctly, you were saying that support for `OPENTHREAD_LINK_METRICS_INITIATOR` should stay in FTD.  Do you think we can remove it then or are there some other reasons to leave it?

Reply:
I think we can remove it for FTD as well if its not required. I was thinking more about Initiator role in general.

Reply:
removed